app/src/main/java/com/example/cardgame/GameplayFragment.kt

```diff
@@ -0,0 +1,393 @@
+
+                    val nextCard = returnCard
+                    Log.d(TAG, "${cardValues[nextCard]} < ${cardValues[currentCard]}")
+                    if(nextCard < currentCard) {
+                        currentCard = nextCard
+                        currentScore++
+
+                        activity?.runOnUiThread {
+                            val imgView = root!!.findViewById<ImageView>(R.id.ivCurrentCard)
+                            Picasso.get().load(image).into(imgView)
+                            val tv = root!!.findViewById<TextView>(R.id.tvScore)
+                            tv.text = currentScore.toString()
+                        }
+                    }
+                    else endGame()
+                    currentCard = nextCard
+                }
+                else {
+                    Log.d(TAG, "RESPONSE FOR NEW CARD IS UNSUCCESSFUL")
+                }
+            }
+        })
+
+    }
+
+    fun higherCard(btn : View) {
+        val client = OkHttpClient()
+        val deckApi = "https://deckofcardsapi.com/api/deck/$deckId/draw/?count=1"
+        val request = Request.Builder().url(deckApi).build()
+
+        client.newCall(request).enqueue(object : Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                e.printStackTrace()
+                Log.d(TAG, "REQUEST FOR NEW CARD FAILED!" + e.printStackTrace())
+            }
+
+            override fun onResponse(call: Call, response: Response) {
+                if(response.isSuccessful) {
+                    val result = response.body()!!.string()
+                    val strRes = Gson().fromJson<ResponseForNewCard>(result,
+                        ResponseForNewCard::class.java)
+                    val image = strRes.cards[0].image
+                    currentImage = image
+                    val cardValue = strRes.cards[0].value
+                    returnCard = cardValues.indexOf(cardValue)
+
+                    val nextCard = returnCard
+                    Log.d(TAG, "${cardValues[nextCard]} > ${cardValues[currentCard]}")
+                    if(nextCard > currentCard) {
+                        currentCard = nextCard
+                        currentScore++
+
+                        activity?.runOnUiThread {
+                            val imgView = root!!.findViewById<ImageView>(R.id.ivCurrentCard)
+                            Picasso.get().load(image).into(imgView)
+                            val tv = root!!.findViewById<TextView>(R.id.tvScore)
+                            tv.text = currentScore.toString()
+                        }
+                    }
+                    else endGame()
+
+                    currentCard = nextCard
+
+                }
+                else {
+                    Log.d(TAG, "RESPONSE FOR NEW CARD IS UNSUCCESSFUL")
+                }
+            }
+        })
+    }
+
+    //toDo izbrisi ovu metodu
+    private fun callApiForNextCard(root: View): Int {
```

Comment:
This can be used instead of `drawNewCard()` because it returns the card that can be used to reduce duplicated code in `higherCard()` and `lowerCard()`.